examples/presenting_revoked_credential_anoncreds/example.py

```diff
@@ -0,0 +1,186 @@
+"""Minimal reproducible example script.
+
+This script is for you to use to reproduce a bug or demonstrate a feature.
+"""
+
+import asyncio
+import json
+from os import getenv
+import time
+
+from acapy_controller import Controller
+from acapy_controller.logging import logging_to_stdout, section
+from acapy_controller.models import V20PresExRecord, V20PresExRecordList
+from acapy_controller.protocols import (
+    didexchange,
+    indy_anoncred_credential_artifacts,
+    indy_anoncred_onboard,
+    indy_anoncreds_publish_revocation,
+    indy_anoncreds_revoke,
+    indy_issue_credential_v2,
+    indy_present_proof_v2,
+)
+
+ALICE = getenv("ALICE", "http://alice:3001")
+BOB = getenv("BOB", "http://bob:3001")
+
+
+def summary(presentation: V20PresExRecord) -> str:
+    """Summarize a presentation exchange record."""
+    request = presentation.pres_request
+    return "Summary: " + json.dumps(
+        {
+            "state": presentation.state,
+            "verified": presentation.verified,
+            "presentation_request": request.dict(by_alias=True) if request else None,
+        },
+        indent=2,
+        sort_keys=True,
+    )
+
+
+async def main():
+    """Test Controller protocols."""
+    async with Controller(base_url=ALICE) as alice, Controller(base_url=BOB) as bob:
+        # Connecting
+        with section("Establish connection"):
+            alice_conn, bob_conn = await didexchange(alice, bob)
+
+        with section("Create and publish schema and cred def"):
+            # Issuance prep
+            alice_did = await indy_anoncred_onboard(alice)
+            schema, cred_def = await indy_anoncred_credential_artifacts(
+                alice,
+                ["firstname", "lastname"],
+                support_revocation=True,
+                anoncreds_wallet=True,
+                issuerID=alice_did.did,
+            )
+
+        with section("Issue credential to Bob"):
+            # Issue a credential
+            alice_cred_ex, _ = await indy_issue_credential_v2(
+                alice,
+                bob,
+                alice_conn.connection_id,
+                bob_conn.connection_id,
+                cred_def["credential_definition_id"],
```

Comment:
I'd like to be able to use `cred_def.credential_definition_id` rather than having to use dictionary access. I think that means we'd have to make the `CredDefResultAnonCreds.schema_state` be itself a Minimal dataclass with attributes on it.

With the way deserialization is handled, that means we'd have to define a class method `deserialize` on the `CredDefResultAnonCreds` like this one on the Invitation dataclass:

https://github.com/Indicio-tech/acapy-minimal-example/blob/ed7e1874c1c26bb8e92e40ac9b7424a2fb84f3da/acapy_controller/protocols.py#L146-L151

Reply:
Similar comments for `SchemaResultAnonCreds`